amd/src/mod_hsuforum_focus.js

```diff
@@ -33,6 +33,24 @@
                     }
                 }
             });
+
+            $('.expandalldiscussions').on('click', function () {
+                if ($('.expandalldiscussions').hasClass('expanded')) {
+                    $('.expandalldiscussions')[0].innerHTML = 'Expand All'
+                    const collapsable = $('.posts-collapse-container.collapse');
+                    for (let x = 0; x < collapsable.length; x++) {
+                        $('#' + collapsable[x].id).removeClass('show')
+                        $('.expandalldiscussions').removeClass('expanded')
+                    }
+                } else {
+                    $('.expandalldiscussions')[0].innerHTML = 'Collapse All'
+                    const collapsable = $('.posts-collapse-container.collapse');
+                    for (let x = 0; x < collapsable.length; x++) {
+                        $('#' + collapsable[x].id).addClass('show')
+                        $('.expandalldiscussions').addClass('expanded')
+                    }
```

Comment:
Did you try my suggestion in the implementation plan? Think that does the same with less code.

Reply:
I used your function, still needs to be run on every toggable element no the page so no real change in code length